src/ert/gui/model/node.py

```diff
@@ -9,28 +17,111 @@ class NodeType(Enum):
     JOB = auto()
 
 
-class Node:
-    def __init__(self, id_: int, data: Dict, type_: NodeType) -> None:
-        self.parent: Optional[Node] = None
-        self.data: Dict = data
-        self.children: Dict[int, Node] = {}
-        self.id = id_
-        self.type = type_
+@dataclass
+class _Node(ABC):
+    id_: int
+    data: Dict[Any, Any] = field(default_factory=dict)
+    parent: Optional[Union[RootNode, IterNode, RealNode]] = None
+    children: Dict[int, Union[IterNode, RealNode, ForwardModelNode]] = field(
+        default_factory=dict
+    )
 
     def __repr__(self) -> str:
         parent = "no " if self.parent is None else ""
         children = "no " if len(self.children) == 0 else f"{len(self.children)} "
-        return f"Node<{self.type}>@{self.id} with {parent}parent and {children}children"
+        return (
+            f"Node<{type(self)}>@{self.id_} with {parent}parent and {children}children"
+        )
+
+    @abstractmethod
+    def add_child(
+        self,
+        node: Union[IterNode, RealNode, ForwardModelNode],
+        node_id: Optional[int] = None,
+    ) -> None:
+        raise NotImplementedError()
+
+    def row(self) -> int:
+        if "index" in self.data:
+            return int(self.data["index"])
+        if self.parent:
+            return list(self.parent.children.keys()).index(self.id_)
+        raise ValueError(f"{self} had no parent")
 
-    def add_child(self, node: "Node", node_id: Optional[int] = None) -> None:
+
+@dataclass
+class RootNode(_Node):
+    parent: None = field(default=None, init=False)
+    children: Dict[int, IterNode] = field(default_factory=dict)
+
+    def add_child(self, node: IterNode, node_id: Optional[int] = None) -> None:
         node.parent = self
         if node_id is None:
-            node_id = node.id
+            node_id = node.id_
+        self.children[node_id] = node
+
+
+@dataclass
+class IterNodeData:
+    index: Optional[str] = None
+    status: Optional[str] = None
+    _sorted_real_ids: list[str] = field(default_factory=list)
```

Comment:
why start the class property with an underscore? Is it in some way internal to `IterNodeData`?

Reply:
That was the way it was before, but then it was a string which we referred to by a variable `SORTED_REALIZATION_IDS`. Should I change the field to `sorted_realization_ids`?

Reply:
Yes, since this is referred to from snapshot.py it should be "public" thing from IterNode. So we can remove the leading underscore (and similarly for other properties in this PR).

Reply:
I will give the fields descriptive names.